crates/bevy_utils/src/option_ext.rs

```diff
@@ -0,0 +1,90 @@
+//! Extensions to the [`Option`] type used in Bevy.
```

Comment:
I also think the odds of these being included in `std` are slim. And `assume`, once Bevy's error type isn't just a boxed core Error, couldn't possibly live in `std`. And even if `unpack` were to land in std, it would likely be closer to the "years" timeframe than the "months" timeframe.